### PR TITLE
Show an error message if no medicines are added on the Rx screen.

### PR DESCRIPTION
**Pull request by msasikanth**
https://app.clubhouse.io/simpledotorg/story/1429/show-an-error-message-if-no-medicines-are-added-on-the-rx-screen

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdateTest.kt [hunk 1]
```diff
@@ -52,14 +52,34 @@ class TeleconsultPrescriptionUpdateTest {
...
   }
 
+  @Test
+  fun `when there are no medicines added, then show medicines required error`() {
+    val medicalInstructions = "This is a medical instructions"
+    val medicalRegistrationId = "ABC12345"
+
+    updateSpec
+        .given(model)
+        .whenEvent(DataForNextClickLoaded(
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdateTest.kt:71]: Can we add `DataForNextClickLoaded` event as a local property for better readability?

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdate.kt [hunk 2]
```diff
@@ -31,21 +32,27 @@ class TeleconsultPrescriptionUpdate : Update<TeleconsultPrescriptionModel, Telec
...
-              medicalInstructions = event.medicalInstructions
-          )
-      )
-    } else {
-      dispatch(ShowSignatureRequiredError)
+    return when {
+      event.hasSignatureBitmap.not() -> {
+        dispatch(ShowSignatureRequiredError)
+      }
+      event.hasMedicines.not() -> {
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdate.kt:39]: Error scenario : 
Step 1: There's a possibility where the medical officer has not added both Signature and Medicines. In this case, when I click on `Next` button, only signature error is enabled.
Step 2: I add the signature then click on `Next`, we get the medicine error. 

Expected behaviour: On click of `Next`, we should be getting both the errors together if empty.

**Review by Honey14 — CHANGES_REQUESTED** [review]
Looks great. Left few comments. PTAL

File: app/src/test/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdateTest.kt (hunk 1 above)
Comment (msasikanth) [reply app/src/test/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdateTest.kt:71]: we can, but we are not reusing it anywhere and IMO the implementation is not large enough to extract to an variable 🤔

File: app/src/main/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdate.kt (hunk 2 above)
Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdate.kt:39]: Ah good catch, will add this.

File: app/src/test/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdateTest.kt (hunk 1 above)
Comment (Honey14) [reply app/src/test/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionUpdateTest.kt:71]: Sure, no problem. Had readability in mind., but this works as well.

**Review by Honey14 — APPROVED** [review]
Looks good

**Pull request merged**